{
  message: {
    function: 'AdsController.createAd',
    error: TypeError: Cannot read properties of undefined (reading 'adsService')
        at C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:12:35
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:4:12)
        at createAd (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:21:16)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at C:\Users\aabadin\Documents\code\informe5Backend\middlewares\zod.validate.ts:23:5
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.createAd',
    error: TypeError: Cannot read properties of undefined (reading 'adsService')
        at C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:12:35
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:4:12)
        at createAd (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:22:16)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at C:\Users\aabadin\Documents\code\informe5Backend\middlewares\zod.validate.ts:23:5
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.createAd',
    error: TypeError: Cannot read properties of undefined (reading 'service')
        at C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:12:35
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:4:12)
        at createAd (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:22:16)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at C:\Users\aabadin\Documents\code\informe5Backend\middlewares\zod.validate.ts:22:5
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.createAd',
    error: TypeError: this.service.createAd is not a function
        at AdsController.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:17:43)
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:4:12)
        at AdsController.createAd (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:24:16)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at C:\Users\aabadin\Documents\code\informe5Backend\middlewares\zod.validate.ts:22:5
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.createAd',
    error: TypeError: this.service.createAd is not a function
        at AdsController.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:17:43)
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:4:12)
        at AdsController.createAd (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:24:16)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at C:\Users\aabadin\Documents\code\informe5Backend\middlewares\zod.validate.ts:22:5
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.createAd',
    error: TypeError: adsServiceLoad.createAd is not a function
        at AdsController.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:17:45)
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:4:12)
        at AdsController.createAd (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:24:16)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at C:\Users\aabadin\Documents\code\informe5Backend\middlewares\zod.validate.ts:22:5
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsService.createAd',
    error: PrismaClientValidationError: 
    Invalid `this.prisma.ads.create()` invocation in
    C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.service.ts:10:48
    
       7 constructor (
       8   public createAd = async <T>(data: createAdType): Promise<GenericResponseObject<T | null>> => {
       9     try {
    â†’ 10       const response = await this.prisma.ads.create({
                 data: {
                   importance: "1",
                               ~~~
                   photoUrl: "public\\image-1705134774367-757782038-descarga.jfif",
                   title: "Primera Ad",
                   url: "http://adrian.com",
                   user: {
                     connect: {
                       id: "4c2b66ed-b656-41de-96d0-664c25f9ba55"
                     }
                   }
                 }
               })
    
    Argument `importance`: Invalid value provided. Expected Int, provided String.
        at ti (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:118:5888)
        at si.handleRequestError (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6473)
        at si.handleAndLogRequestError (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6151)
        at si.request (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:5859)
        at l (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:130:10025) {
      clientVersion: '5.7.1'
    }
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.getAds',
    error: TypeError: Cannot read properties of undefined (reading 'service')
        at C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:28:35
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:4:12)
        at getAds (C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.controller.ts:37:16)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at Route.dispatch (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:114:3)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.setActive',
    error: TypeError: Cannot read properties of undefined (reading 'service')
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:39:35
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:4:12)
        at setActive (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:51:16)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at Route.dispatch (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:114:3)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'schemaValidator',
    error: ZodError: [
      {
        "code": "too_small",
        "minimum": 3,
        "type": "string",
        "inclusive": true,
        "exact": false,
        "message": "Debe se una cadena de al menos 3 caracteres",
        "path": [
          "body",
          "usersId"
        ]
      }
    ]
        at Object.get error [as error] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\zod\lib\types.js:43:31)
        at ZodObject.parse (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\zod\lib\types.js:143:22)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\middlewares\zod.validate.ts:16:14
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at done (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\multer\lib\make-middleware.js:45:7)
        at indicateDone (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\multer\lib\make-middleware.js:49:68)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\multer\lib\make-middleware.js:155:11
        at WriteStream.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\multer\storage\disk.js:43:9)
        at WriteStream.emit (node:events:526:35) {
      issues: [
        {
          code: 'too_small',
          minimum: 3,
          type: 'string',
          inclusive: true,
          exact: false,
          message: 'Debe se una cadena de al menos 3 caracteres',
          path: [ 'body', 'usersId' ]
        }
      ],
      addIssue: [Function (anonymous)],
      addIssues: [Function (anonymous)],
      errors: [
        {
          code: 'too_small',
          minimum: 3,
          type: 'string',
          inclusive: true,
          exact: false,
          message: 'Debe se una cadena de al menos 3 caracteres',
          path: [ 'body', 'usersId' ]
        }
      ]
    }
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.updateAd',
    error: TypeError: Cannot destructure property 'filename' of 'req.file' as it is undefined.
        at AdsController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:93:15)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:4:12)
        at AdsController.updateAd (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:108:16)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at Route.dispatch (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:114:3)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.updateAd',
    error: TypeError: Cannot destructure property 'filename' of 'req.file' as it is undefined.
        at AdsController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:93:15)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:4:12)
        at AdsController.updateAd (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:108:16)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at Route.dispatch (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:114:3)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.updateAd',
    error: TypeError: Cannot destructure property 'filename' of 'req.file' as it is undefined.
        at AdsController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:93:15)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:4:12)
        at AdsController.updateAd (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:108:16)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at Route.dispatch (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:114:3)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.updateAd',
    error: TypeError: Cannot destructure property 'filename' of 'req.file' as it is undefined.
        at AdsController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:93:15)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:4:12)
        at AdsController.updateAd (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:108:16)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at Route.dispatch (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:114:3)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.updateAd',
    error: TypeError: Cannot destructure property 'filename' of 'req.file' as it is undefined.
        at AdsController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:93:15)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:4:12)
        at AdsController.updateAd (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:108:16)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at Route.dispatch (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:114:3)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.updateAd',
    error: TypeError: Cannot destructure property 'filename' of 'req.file' as it is undefined.
        at AdsController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:93:15)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:4:12)
        at AdsController.updateAd (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:108:16)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at Route.dispatch (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:114:3)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.updateAd',
    error: TypeError: Cannot destructure property 'filename' of 'req.file' as it is undefined.
        at AdsController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:93:15)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:4:12)
        at AdsController.updateAd (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:108:16)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at Route.dispatch (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:114:3)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsController.updateAd',
    error: TypeError: Cannot destructure property 'filename' of 'req.file' as it is undefined.
        at AdsController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:93:15)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:4:12)
        at AdsController.updateAd (C:\Users\Adria\Documents\code\turco2\informe5Backend\ads\ads.controller.ts:108:16)
        at Layer.handle [as handle_request] (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at done (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\multer\lib\make-middleware.js:45:7)
        at indicateDone (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\multer\lib\make-middleware.js:49:68)
  },
  level: 'error'
}
{
  message: {
    function: 'AdsService.setActive',
    error: PrismaClientValidationError: 
    Invalid `this.prisma.ads.update()` invocation in
    C:\Users\aabadin\Documents\code\informe5Backend\ads\ads.service.ts:29:46
    
      26 },
      27 public setActive = async (id: string) => {
      28   try {
    â†’ 29     const result = await this.prisma.ads.update({
               where: {
                 id: "550a1bae-9815-4ebd-894f-7179ae6be067"
               },
               data: {
                 isActive: true,
                 ~~~~~~~~
             ?   id?: String | StringFieldUpdateOperationsInput,
             ?   createdAt?: DateTime | DateTimeFieldUpdateOperationsInput,
             ?   updatedAt?: DateTime | DateTimeFieldUpdateOperationsInput,
             ?   importance?: Int | IntFieldUpdateOperationsInput,
             ?   url?: String | StringFieldUpdateOperationsInput,
             ?   title?: String | StringFieldUpdateOperationsInput,
             ?   photoUrl?: String | StringFieldUpdateOperationsInput,
             ?   metrics?: MetricUpdateManyWithoutAdsNestedInput,
             ?   user?: UsersUpdateOneRequiredWithoutAdsNestedInput
               }
             })
    
    Unknown argument `isActive`. Available options are marked with ?.
        at ti (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:118:5888)
        at si.handleRequestError (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6473)
        at si.handleAndLogRequestError (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6151)
        at si.request (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:5859)
        at l (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:130:10025) {
      clientVersion: '5.7.1'
    }
  },
  level: 'error'
}
{
  message: {
    function: 'schemaValidator',
    error: ZodError: [
      {
        "code": "too_small",
        "minimum": 3,
        "type": "string",
        "inclusive": true,
        "exact": false,
        "message": "Debe se una cadena de al menos 3 caracteres",
        "path": [
          "body",
          "usersId"
        ]
      }
    ]
        at Object.get error [as error] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\zod\lib\types.js:43:31)
        at ZodObject.parse (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\zod\lib\types.js:143:22)
        at C:\Users\aabadin\Documents\code\informe5Backend\middlewares\zod.validate.ts:16:14
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at done (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\multer\lib\make-middleware.js:45:7)
        at indicateDone (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\multer\lib\make-middleware.js:49:68)
        at Multipart.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\multer\lib\make-middleware.js:166:7)
        at Multipart.emit (node:events:514:28)
        at emitCloseNT (node:internal/streams/destroy:132:10) {
      issues: [
        {
          code: 'too_small',
          minimum: 3,
          type: 'string',
          inclusive: true,
          exact: false,
          message: 'Debe se una cadena de al menos 3 caracteres',
          path: [ 'body', 'usersId' ]
        }
      ],
      addIssue: [Function (anonymous)],
      addIssues: [Function (anonymous)],
      errors: [
        {
          code: 'too_small',
          minimum: 3,
          type: 'string',
          inclusive: true,
          exact: false,
          message: 'Debe se una cadena de al menos 3 caracteres',
          path: [ 'body', 'usersId' ]
        }
      ]
    }
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.jwtLoginVerify',
    error: TypeError: req.cookies is not a function
        at AuthService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:82:15)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:5:58)
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.jwtLoginVerify',
    error: TypeError: req.cookies is not a function
        at AuthService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:82:15)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:5:58)
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.jwtLoginVerify',
    error: TypeError: req.cookies is not a function
        at AuthService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:82:15)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:5:58)
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.jwtLoginVerify',
    error: TypeError: req.cookies is not a function
        at AuthService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:82:15)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:5:58)
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.jwtLoginVerify',
    error: TypeError: req.cookies is not a function
        at AuthService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:82:15)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:5:58)
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.jwtLoginVerify',
    error: TypeError: req.cookies is not a function
        at AuthService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:82:15)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:5:58)
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.jwtLoginVerify',
    error: TypeError: req.cookies is not a function
        at AuthService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:82:15)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:5:58)
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.jwtLoginVerify',
    error: TypeError: req.cookies is not a function
        at AuthService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:82:15)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:5:58)
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.jwtLoginVerify',
    error: TypeError: req.cookies is not a function
        at AuthService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:82:15)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:5:58)
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.jwtLoginVerify',
    error: TypeError: req.cookies is not a function
        at AuthService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:82:15)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:5:58)
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.localLoginVerify',
    error: NotFoundError [PrismaClientKnownRequestError]: No Users found
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:34:610
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:34:550 {
      code: 'P2025',
      clientVersion: '5.7.1',
      meta: undefined
    }
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.localLoginVerify',
    error: NotFoundError [PrismaClientKnownRequestError]: No Users found
        at C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:34:610
        at C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:34:550 {
      code: 'P2025',
      clientVersion: '5.7.1',
      meta: undefined
    }
  },
  level: 'error'
}
{
  message: {
    function: 'DatabaseHandler.FindById',
    error: NotFoundError [PrismaClientKnownRequestError]: No Users found
        at C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:34:610
        at C:\Users\aabadin\Documents\code\informe5Backend\node_modules\@prisma\client\runtime\library.js:34:550 {
      code: 'P2025',
      clientVersion: '5.7.1',
      meta: undefined
    }
  },
  level: 'error'
}
{
  message: {
    function: 'AuthService.jwtLoginVerify',
    error: TypeError: Cannot read properties of null (reading 'accessToken')
        at AuthService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:74:44)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.service.ts:5:58)
  },
  level: 'error'
}
{
  message: {
    function: 'AuthController.innerToken',
    error: GaxiosError: invalid_grant
        at Gaxios._request (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\gaxios\src\gaxios.ts:172:15)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at OAuth2Client.getTokenAsync (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\google-auth-library\build\src\auth\oauth2client.js:122:21) {
      config: {
        method: 'POST',
        url: 'https://oauth2.googleapis.com/token',
        data: '<<REDACTED> - See `errorRedactor` option in `gaxios` for configuration>.',
        headers: {
          'Content-Type': 'application/x-www-form-urlencoded',
          'User-Agent': 'google-api-nodejs-client/9.5.0',
          'x-goog-api-client': 'gl-node/20.10.0'
        },
        paramsSerializer: [Function: paramsSerializer],
        body: '<<REDACTED> - See `errorRedactor` option in `gaxios` for configuration>.',
        validateStatus: [Function: validateStatus],
        responseType: 'unknown',
        errorRedactor: [Function: defaultErrorRedactor]
      },
      response: {
        config: {
          method: 'POST',
          url: 'https://oauth2.googleapis.com/token',
          data: '<<REDACTED> - See `errorRedactor` option in `gaxios` for configuration>.',
          headers: {
            'Content-Type': 'application/x-www-form-urlencoded',
            'User-Agent': 'google-api-nodejs-client/9.5.0',
            'x-goog-api-client': 'gl-node/20.10.0'
          },
          paramsSerializer: [Function: paramsSerializer],
          body: '<<REDACTED> - See `errorRedactor` option in `gaxios` for configuration>.',
          validateStatus: [Function: validateStatus],
          responseType: 'unknown',
          errorRedactor: [Function: defaultErrorRedactor]
        },
        data: { error: 'invalid_grant', error_description: 'Bad Request' },
        headers: {
          'alt-svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000',
          'cache-control': 'no-cache, no-store, max-age=0, must-revalidate',
          connection: 'close',
          'content-encoding': 'gzip',
          'content-type': 'application/json; charset=utf-8',
          date: 'Sun, 28 Jan 2024 19:31:39 GMT',
          expires: 'Mon, 01 Jan 1990 00:00:00 GMT',
          pragma: 'no-cache',
          server: 'scaffolding on HTTPServer2',
          'transfer-encoding': 'chunked',
          vary: 'Origin, X-Origin, Referer',
          'x-content-type-options': 'nosniff',
          'x-frame-options': 'SAMEORIGIN',
          'x-xss-protection': '0'
        },
        status: 400,
        statusText: 'Bad Request',
        request: { responseURL: 'https://oauth2.googleapis.com/token' }
      },
      error: undefined,
      status: 400
    }
  },
  level: 'error'
}
{
  message: {
    function: 'GoogleService.folderExists',
    error: Error: No access, refresh token, API key or refresh handler callback is set.
        at OAuth2Client.getRequestMetadataAsync (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\google-auth-library\build\src\auth\oauth2client.js:269:19)
        at OAuth2Client.requestAsync (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\google-auth-library\build\src\auth\oauth2client.js:368:34)
        at OAuth2Client.request (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\google-auth-library\build\src\auth\oauth2client.js:362:25)
        at createAPIRequestAsync (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\googleapis-common\build\src\apirequest.js:273:31)
        at createAPIRequest (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\googleapis-common\build\src\apirequest.js:52:16)
        at Resource$Files.list (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\googleapis\build\src\apis\drive\v3.js:838:65)
        at GoogleService.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\Services\google.service.ts:31:50)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\Services\google.service.ts:8:71
        at new Promise (<anonymous>)
  },
  level: 'error'
}
{
  message: {
    function: 'postController.eraseAudio',
    error: PrismaClientKnownRequestError: 
    Invalid `this.prisma.audio.delete()` invocation in
    C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:425:55
    
      422 public eraseAudio = async (req: Request<any, any, any, { id: string }>, res: Response) => {
      423   try {
      424     const { id } = req.query
    â†’ 425     const driveId = await this.prisma.audio.delete(
    An operation failed because it depends on one or more records that were required but not found. Record to delete does not exist.
        at si.handleRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6817)
        at si.handleAndLogRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6151)
        at si.request (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:5859)
        at l (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:130:10025) {
      code: 'P2025',
      clientVersion: '5.7.1',
      meta: { modelName: 'Audio', cause: 'Record to delete does not exist.' }
    }
  },
  level: 'error'
}
{
  message: {
    function: 'postController.eraseAudio',
    error: PrismaClientKnownRequestError: 
    Invalid `this.prisma.audio.delete()` invocation in
    C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:425:55
    
      422 public eraseAudio = async (req: Request<any, any, any, { id: string }>, res: Response) => {
      423   try {
      424     const { id } = req.query
    â†’ 425     const driveId = await this.prisma.audio.delete(
    An operation failed because it depends on one or more records that were required but not found. Record to delete does not exist.
        at si.handleRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6817)
        at si.handleAndLogRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6151)
        at si.request (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:5859)
        at l (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:130:10025) {
      code: 'P2025',
      clientVersion: '5.7.1',
      meta: { modelName: 'Audio', cause: 'Record to delete does not exist.' }
    }
  },
  level: 'error'
}
{
  message: {
    function: 'postController.eraseAudio',
    error: PrismaClientKnownRequestError: 
    Invalid `this.prisma.audio.delete()` invocation in
    C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:425:55
    
      422 public eraseAudio = async (req: Request<any, any, any, { id: string }>, res: Response) => {
      423   try {
      424     const { id } = req.query
    â†’ 425     const driveId = await this.prisma.audio.delete(
    An operation failed because it depends on one or more records that were required but not found. Record to delete does not exist.
        at si.handleRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6817)
        at si.handleAndLogRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6151)
        at si.request (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:5859)
        at l (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:130:10025) {
      code: 'P2025',
      clientVersion: '5.7.1',
      meta: { modelName: 'Audio', cause: 'Record to delete does not exist.' }
    }
  },
  level: 'error'
}
{
  message: {
    function: 'postController.eraseAudio',
    error: PrismaClientKnownRequestError: 
    Invalid `this.prisma.audio.delete()` invocation in
    C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:425:55
    
      422 public eraseAudio = async (req: Request<any, any, any, { id: string }>, res: Response) => {
      423   try {
      424     const { id } = req.query
    â†’ 425     const driveId = await this.prisma.audio.delete(
    An operation failed because it depends on one or more records that were required but not found. Record to delete does not exist.
        at si.handleRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6817)
        at si.handleAndLogRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6151)
        at si.request (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:5859)
        at l (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:130:10025) {
      code: 'P2025',
      clientVersion: '5.7.1',
      meta: { modelName: 'Audio', cause: 'Record to delete does not exist.' }
    }
  },
  level: 'error'
}
{
  message: {
    function: 'postController.eraseAudio',
    error: PrismaClientKnownRequestError: 
    Invalid `this.prisma.audio.delete()` invocation in
    C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:425:55
    
      422 public eraseAudio = async (req: Request<any, any, any, { id: string }>, res: Response) => {
      423   try {
      424     const { id } = req.query
    â†’ 425     const driveId = await this.prisma.audio.delete(
    An operation failed because it depends on one or more records that were required but not found. Record to delete does not exist.
        at si.handleRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6817)
        at si.handleAndLogRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6151)
        at si.request (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:5859)
        at l (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:130:10025) {
      code: 'P2025',
      clientVersion: '5.7.1',
      meta: { modelName: 'Audio', cause: 'Record to delete does not exist.' }
    }
  },
  level: 'error'
}
{
  message: {
    function: 'postController.eraseAudio',
    error: PrismaClientKnownRequestError: 
    Invalid `this.prisma.audio.delete()` invocation in
    C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:425:55
    
      422 public eraseAudio = async (req: Request<any, any, any, { id: string }>, res: Response) => {
      423   try {
      424     const { id } = req.query
    â†’ 425     const driveId = await this.prisma.audio.delete(
    An operation failed because it depends on one or more records that were required but not found. Record to delete does not exist.
        at si.handleRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6817)
        at si.handleAndLogRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6151)
        at si.request (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:5859)
        at l (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:130:10025) {
      code: 'P2025',
      clientVersion: '5.7.1',
      meta: { modelName: 'Audio', cause: 'Record to delete does not exist.' }
    }
  },
  level: 'error'
}
{
  message: {
    function: 'postController.eraseAudio',
    error: PrismaClientKnownRequestError: 
    Invalid `this.prisma.audio.delete()` invocation in
    C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:425:55
    
      422 public eraseAudio = async (req: Request<any, any, any, { id: string }>, res: Response) => {
      423   try {
      424     const { id } = req.query
    â†’ 425     const driveId = await this.prisma.audio.delete(
    An operation failed because it depends on one or more records that were required but not found. Record to delete does not exist.
        at si.handleRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6817)
        at si.handleAndLogRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6151)
        at si.request (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:5859)
        at l (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:130:10025) {
      code: 'P2025',
      clientVersion: '5.7.1',
      meta: { modelName: 'Audio', cause: 'Record to delete does not exist.' }
    }
  },
  level: 'error'
}
{
  message: {
    function: 'DatabaseHandler.GetAll',
    error: PrismaClientKnownRequestError: 
    Invalid `prisma.posts.findMany()` invocation:
    
    
    Server has closed the connection.
        at si.handleRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6817)
        at si.handleAndLogRequestError (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:6151)
        at si.request (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:125:5859)
        at l (C:\Users\Adria\Documents\code\turco2\informe5Backend\node_modules\@prisma\client\runtime\library.js:130:10025) {
      code: 'P1017',
      clientVersion: '5.7.1',
      meta: { modelName: 'Posts' }
    }
  },
  level: 'error'
}
{
  message: {
    function: 'schemaValidator',
    error: ZodError: [
      {
        "code": "invalid_type",
        "expected": "array",
        "received": "string",
        "path": [
          "body",
          "audio"
        ],
        "message": "Expected array, received string"
      }
    ]
        at Object.get error [as error] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\zod\lib\types.js:43:31)
        at ZodObject.parse (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\zod\lib\types.js:143:22)
        at C:\Users\aabadin\Documents\code\informe5Backend\middlewares\zod.validate.ts:16:14
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at jwtRenewalToken (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.controller.ts:38:11)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at complete (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\passport\lib\middleware\authenticate.js:271:13)
        at C:\Users\aabadin\Documents\code\informe5Backend\node_modules\passport\lib\middleware\authenticate.js:278:15 {
      issues: [
        {
          code: 'invalid_type',
          expected: 'array',
          received: 'string',
          path: [ 'body', 'audio' ],
          message: 'Expected array, received string'
        }
      ],
      addIssue: [Function (anonymous)],
      addIssues: [Function (anonymous)],
      errors: [
        {
          code: 'invalid_type',
          expected: 'array',
          received: 'string',
          path: [ 'body', 'audio' ],
          message: 'Expected array, received string'
        }
      ]
    }
  },
  level: 'error'
}
{
  message: {
    function: 'schemaValidator',
    error: ZodError: [
      {
        "code": "invalid_type",
        "expected": "array",
        "received": "string",
        "path": [
          "body",
          "audio"
        ],
        "message": "Expected array, received string"
      }
    ]
        at Object.get error [as error] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\zod\lib\types.js:43:31)
        at ZodObject.parse (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\zod\lib\types.js:143:22)
        at C:\Users\aabadin\Documents\code\informe5Backend\middlewares\zod.validate.ts:16:14
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at jwtRenewalToken (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.controller.ts:38:11)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at complete (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\passport\lib\middleware\authenticate.js:271:13)
        at C:\Users\aabadin\Documents\code\informe5Backend\node_modules\passport\lib\middleware\authenticate.js:278:15 {
      issues: [
        {
          code: 'invalid_type',
          expected: 'array',
          received: 'string',
          path: [ 'body', 'audio' ],
          message: 'Expected array, received string'
        }
      ],
      addIssue: [Function (anonymous)],
      addIssues: [Function (anonymous)],
      errors: [
        {
          code: 'invalid_type',
          expected: 'array',
          received: 'string',
          path: [ 'body', 'audio' ],
          message: 'Expected array, received string'
        }
      ]
    }
  },
  level: 'error'
}
{
  message: {
    function: 'schemaValidator',
    error: ZodError: [
      {
        "code": "invalid_type",
        "expected": "array",
        "received": "string",
        "path": [
          "body",
          "audio"
        ],
        "message": "Expected array, received string"
      }
    ]
        at Object.get error [as error] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\zod\lib\types.js:43:31)
        at ZodObject.parse (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\zod\lib\types.js:143:22)
        at C:\Users\aabadin\Documents\code\informe5Backend\middlewares\zod.validate.ts:16:14
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at jwtRenewalToken (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.controller.ts:38:11)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at complete (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\passport\lib\middleware\authenticate.js:271:13)
        at C:\Users\aabadin\Documents\code\informe5Backend\node_modules\passport\lib\middleware\authenticate.js:278:15 {
      issues: [
        {
          code: 'invalid_type',
          expected: 'array',
          received: 'string',
          path: [ 'body', 'audio' ],
          message: 'Expected array, received string'
        }
      ],
      addIssue: [Function (anonymous)],
      addIssues: [Function (anonymous)],
      errors: [
        {
          code: 'invalid_type',
          expected: 'array',
          received: 'string',
          path: [ 'body', 'audio' ],
          message: 'Expected array, received string'
        }
      ]
    }
  },
  level: 'error'
}
{
  message: {
    function: 'schemaValidator',
    error: ZodError: [
      {
        "code": "custom",
        "message": "Invalid input",
        "path": [
          "body",
          "audio"
        ]
      }
    ]
        at Object.get error [as error] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\zod\lib\types.js:43:31)
        at ZodObject.parse (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\zod\lib\types.js:143:22)
        at C:\Users\aabadin\Documents\code\informe5Backend\middlewares\zod.validate.ts:16:14
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at jwtRenewalToken (C:\Users\aabadin\Documents\code\informe5Backend\auth\auth.controller.ts:38:11)
        at Layer.handle [as handle_request] (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\layer.js:95:5)
        at next (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\express\lib\router\route.js:144:13)
        at complete (C:\Users\aabadin\Documents\code\informe5Backend\node_modules\passport\lib\middleware\authenticate.js:271:13)
        at C:\Users\aabadin\Documents\code\informe5Backend\node_modules\passport\lib\middleware\authenticate.js:278:15 {
      issues: [
        {
          code: 'custom',
          message: 'Invalid input',
          path: [ 'body', 'audio' ]
        }
      ],
      addIssue: [Function (anonymous)],
      addIssues: [Function (anonymous)],
      errors: [
        {
          code: 'custom',
          message: 'Invalid input',
          path: [ 'body', 'audio' ]
        }
      ]
    }
  },
  level: 'error'
}
{ message: { imagesArray: [] }, level: 'error' }
{
  message: {
    function: 'facebookService.getLinkFromId',
    error: Error: Error creating link from id array
        at FacebookService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:108:17)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:5:58)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
  },
  level: 'error'
}
{ message: { imagesArray: [] }, level: 'error' }
{
  message: {
    function: 'facebookService.getLinkFromId',
    error: Error: Error creating link from id array
        at FacebookService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:108:17)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:5:58)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
  },
  level: 'error'
}
{ message: { imagesArray: [] }, level: 'error' }
{
  message: {
    function: 'facebookService.getLinkFromId',
    error: Error: Error creating link from id array
        at FacebookService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:108:17)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:5:58)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
  },
  level: 'error'
}
{ message: { imagesArray: [] }, level: 'error' }
{
  message: {
    function: 'facebookService.getLinkFromId',
    error: Error: Error creating link from id array
        at FacebookService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:108:17)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:5:58)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
  },
  level: 'error'
}
{ message: { imagesArray: [] }, level: 'error' }
{
  message: {
    function: 'facebookService.getLinkFromId',
    error: Error: Error creating link from id array
        at FacebookService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:108:17)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:5:58)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
  },
  level: 'error'
}
{ message: { imagesArray: [] }, level: 'error' }
{
  message: {
    function: 'facebookService.getLinkFromId',
    error: Error: Error creating link from id array
        at FacebookService.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:108:17)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\aabadin\Documents\code\informe5Backend\Services\facebook.service.ts:5:58)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
  },
  level: 'error'
}
{
  message: {
    function: 'PostController.createPost',
    error: TypeError: audio.map is not a function
        at DatabaseHandler.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:53:60)
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:4:12)
        at DatabaseHandler.createPost (C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:39:125)
        at PostController.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\post\post.controller.ts:119:49)
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\post\post.controller.ts:8:71
        at new Promise (<anonymous>)
  },
  level: 'error'
}
{
  message: {
    function: 'PostController.createPost',
    error: TypeError: _a.map is not a function
        at DatabaseHandler.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:53:72)
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:4:12)
        at DatabaseHandler.createPost (C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:39:125)
        at PostController.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\post\post.controller.ts:119:49)
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\post\post.controller.ts:8:71
        at new Promise (<anonymous>)
  },
  level: 'error'
}
{
  message: {
    function: 'PostController.createPost',
    error: TypeError: _a.map is not a function
        at DatabaseHandler.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:54:72)
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:4:12)
        at DatabaseHandler.createPost (C:\Users\aabadin\Documents\code\informe5Backend\post\post.service.ts:39:125)
        at PostController.<anonymous> (C:\Users\aabadin\Documents\code\informe5Backend\post\post.controller.ts:119:49)
        at Generator.next (<anonymous>)
        at C:\Users\aabadin\Documents\code\informe5Backend\post\post.controller.ts:8:71
        at new Promise (<anonymous>)
  },
  level: 'error'
}
{
  message: {
    function: 'PostController.createPost',
    error: TypeError: _a.map is not a function
        at DatabaseHandler.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:54:72)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:4:12)
        at DatabaseHandler.createPost (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:39:125)
        at PostController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:119:49)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:8:71
        at new Promise (<anonymous>)
  },
  level: 'error'
}
{
  message: {
    function: 'PostController.createPost',
    error: TypeError: _a.map is not a function
        at DatabaseHandler.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:54:72)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:4:12)
        at DatabaseHandler.createPost (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:39:125)
        at PostController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:119:49)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:8:71
        at new Promise (<anonymous>)
  },
  level: 'error'
}
{
  message: {
    function: 'PostController.createPost',
    error: SyntaxError: "[object Object]" is not valid JSON
        at JSON.parse (<anonymous>)
        at DatabaseHandler.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:44:24)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:4:12)
        at DatabaseHandler.createPost (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.service.ts:39:125)
        at PostController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:119:49)
        at Generator.next (<anonymous>)
        at C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:8:71
  },
  level: 'error'
}
{
  message: {
    function: 'PostController.createPost',
    error: Error: Error updating Database
        at PostController.<anonymous> (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:167:24)
        at Generator.next (<anonymous>)
        at fulfilled (C:\Users\Adria\Documents\code\turco2\informe5Backend\post\post.controller.ts:5:58)
  },
  level: 'error'
}
